application/controllers/admin/dataentry.php

```diff
@@ -1172,10 +1172,12 @@ public function editdata($subaction, $id, $surveyid)
                                     $aDataentryoutput .= CHtml::numberField($fname['fieldname'], $idrow[$fname['fieldname']],array('step'=>'any'));
                                 } else {
                                     $aDataentryoutput .= "\t<select name='{$fname['fieldname']}' class='form-control'>\n";
-                                    $aDataentryoutput .= "<option value=''>...</option>\n";
+                                    $aDataentryoutput .= "<option value=''";
+                                    if ($idrow[$fname['fieldname']] == '') {$aDataentryoutput .= " selected"; }
+                                    $aDataentryoutput .= ">...</option>\n";
                                     for ($ii = $minvalue; $ii <= $maxvalue; $ii += $stepvalue) {
                                         $aDataentryoutput .= "<option value='$ii'";
-                                        if ($idrow[$fname['fieldname']] == $ii) {$aDataentryoutput .= " selected"; }
+                                        if ($idrow[$fname['fieldname']] != '' && $idrow[$fname['fieldname']] == $ii) {$aDataentryoutput .= " selected"; }
```

Comment:
must not be !== '' ?

else if $ii === "0" it don't work, no  ?

Reply:
thanks I've updated it now to be more clear